app/src/main/java/com/epfl/beatlink/ui/library/PlaylistOverview.kt

```diff
@@ -209,9 +229,70 @@ fun PlaylistOverviewScreen(
             }
       })
   // Show the description overlay if visible
-  if (showDialog) {
+  if (showDialogOverlay) {
     ViewDescriptionOverlay(
-        onDismissRequest = { showDialog = false },
+        onDismissRequest = { showDialogOverlay = false },
         description = selectedPlaylistState.playlistDescription)
   }
+  /** Show the alert dialog for confirmation of the export of the playlist */
+  if (showDialogExport) {
+    AlertDialog(
+        onDismissRequest = { showDialogExport = false },
+        title = {
+          Text(text = "Export Playlist to Spotify", style = MaterialTheme.typography.titleLarge)
+        },
+        text = {
+          Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+            Text(
+                text =
+                    "Are you sure you want to export this playlist to Spotify? Once exported, the playlist will be removed from the app and cannot be recovered.",
+                style = MaterialTheme.typography.bodyMedium)
+            Spacer(modifier = Modifier.height(8.dp))
+          }
+        },
+        confirmButton = {
+          TextButton(
+              modifier = Modifier.testTag("confirmButton"),
+              onClick = {
+                coroutineScope.launch {
+                  val finalList = playlistViewModel.getFinalListTracks()
+                  if (finalList.isNotEmpty()) {
+                    spotifyViewModel.createBeatLinkPlaylist(
+                        playlistName = selectedPlaylistState.playlistName,
+                        playlistDescription = selectedPlaylistState.playlistDescription,
+                        tracks = finalList,
+                        onResult = { idSpotify ->
+                          Log.d("Spotify", idSpotify.toString())
+                          if (idSpotify != null) {
+                            // Delete playlist only if creation was successful
+                            playlistViewModel.deletePlaylist(selectedPlaylistState.playlistID)
+                            showDialogExport = false
+                            Toast.makeText(
+                                    context, "Playlist exported successfully", Toast.LENGTH_SHORT)
+                                .show()
+                            navigationActions.navigateTo(Screen.LIBRARY)
+                          } else {
```

Comment:
I think you should use the `navigateToAndClearBackStack(screen: String, entriesToClear: Int)` function to block the navigation to the `PlaylistOverviewScreen` of the playlist that just got exported when navigating back via the phone back button. Let me know if the current code works or it needs to be changed.

Reply:
You are absolutely right the back button was returning to the overview of the deleted playlist I will fix that. Thanks